### ISSUE TITLE
Issue 017: Build complete MAE MaskingPlan from config and sample metadata

### ISSUE THREAD

**Issue by ozw4** [ozw4/seis-attr-ssl#27]
# Issue 017: Build complete MAE MaskingPlan from config and sample metadata

## Background

Spatial masking and attribute masking must be combined into one object that the strict MAE dataset/model/training loop can consume. This builder should use explicit masks and must not infer masks from zero-valued data.

## Goal

Implement a high-level MAE masking planner that combines:

```text
- spatial block mask
- visible spatial token mask
- attribute input mask
- attribute target mask
- dropped attribute target mask
- input_attribute_ids
- target_attribute_ids
```

## Suggested files

```text
src/seis_attr_ssl/masking/mae.py
src/seis_attr_ssl/masking/schema.py
src/seis_attr_ssl/masking/__init__.py

tests/test_mae_masking_plan.py
```

## Required API

Add a function similar to:

```python
def build_mae_masking_plan(
    available_attribute_ids: Sequence[int],
    target_valid: np.ndarray,
    local_crop_size_xyz: Sequence[int],
    patch_size_xyz: Sequence[int],
    spatial_mask_ratio: float,
    block_size_tokens_xyz: Sequence[int],
    min_input_attributes: int,
    max_input_attributes: int,
    attribute_dropout_prob: float,
    group_dropout_prob: float,
    attribute_groups: Mapping[str, str],
    rng: np.random.Generator,
) -> MaskingPlan:
    ...
```

The exact signature may differ, but the builder must use the MVP config fields.

## Behavior

- Compute token grid shape from local crop size and patch size.
- Generate spatial mask with `True = masked/reconstruction target`.
- Generate visible mask with `~spatial_mask`.
- Generate attribute input mask using attribute and group dropout.
- `attribute_target_mask` should equal `target_valid` by default.
- `dropped_attribute_mask = attribute_target_mask & ~attribute_input_mask`.
- `input_attribute_ids` should match `np.flatnonzero(attribute_input_mask)`.
- `target_attribute_ids` should match the full MVP registry order.
- Validate the resulting `MaskingPlan` before returning.

## Important MVP rule

The pretrain dataset target is fixed registry order. The input `x` may contain only selected attributes. The builder must therefore make it easy for downstream code to distinguish:

```text
- input attributes visible to encoder
- target attributes available for reconstruction
- valid target attributes withheld from input
```

## Tests

Add tests for:

```text
- default config produces token grid [16, 16, 16]
- spatial mask ratio is approximately 0.75
- visible_spatial_mask is complement of spatial_mask
- target_valid with missing attrs propagates to attribute_target_mask
- dropped_attribute_mask equals target_valid & ~input_mask
- deterministic behavior for same seed
- all returned masks use bool dtype
- validation catches inconsistent masks
```

## Out of scope

Do not implement model patchification here.
Do not implement reconstruction loss here.
Do not implement a PyTorch dataloader collate function unless needed for tests.

## Acceptance criteria

- `build_mae_masking_plan` exists and returns a validated plan.
- Unit tests cover all key mask semantics.
- Existing dataset tests still pass.
- `ruff check .`, `pytest -q`, and `python -m compileall -q src proc tests` pass.
